### DISCUSSION TITLE
Moving the search bar based on screen size?

### DISCUSSION THREAD

**Discussion by Mendanbar** [squidfunk/mkdocs-material#2464] [Q&A]
Hi.  Is there a way to test for screen size in the partials somehow?  I would like to have the search included in the header for mobile devices, but move it to the main content for tablet/desktop (One of the requirements for my project).  Thanks!

**Comment by squidfunk** [comment]
You could add some additional CSS and position the search based on media queries. This is exactly what Material for MkDocs does.

**Reply by mason-c-web** [reply]
I think this is what tried but are running into issues. To expand on what @Mendanbar said: We want a search bar on the page on desktop, and to have it move to the header on mobile. The initial solution was to render both in the DOM and use media queries to `display:none` the appropriate one. However, it looks likes if there are two search bars in the DOM there seems to be a conflict where you can only activate search on the highest one, regardless if it is hidden or not. What attribute(s) do i need to change to get this to work? Or is what we want possible with media queries?

**Reply by mason-c-web** [reply]
My only thought is that we can't have two search bars and need to remove it from the DOM completely, but I am unsure if that outside MKDocs control. 🤷

**Reply by squidfunk** [reply]
Theoretically, two search bars should work. The markup must be identical (more precisely contain the `data-md-component=search` attribute + all of the nested elements) and it must be located outside of `.md-container`. However, note that I never tested this, so YMMV.

**Reply by Mendanbar** [reply]
Can you elaborate a bit on why it must be located outside of `.md-container`?  One of the 2 places we'd like the search bar to appear is inside the main content (in between the side navs, and above the main document).  

On another note, I was trying to implement your original suggestion of just having one search bar and positioning it based on media queries, but was running into z fighting with the header if it exists outside that element.  I think we're hoping to take advantage of some of the special behaviors  (taking up the full screen for example) when in mobile without re-inventing all of the associated CSS.

P.S. Thank you for taking the time to respond :)

**Reply by squidfunk** [reply]
Because this is where it's mounted:

https://github.com/squidfunk/mkdocs-material/blob/f0e87564db5d7f4cb0cc426d56d297b67c854eaa/src/assets/javascripts/bundle.ts#L176-L178

The `control$` observable mounts all components that are outside of the content area. The `content$` observable below mounts all components that are inside the container + the header title. This is necessary for instant loading to work, which will leave all components of the `control$` observable untouched, while re-mounting those in the `content$` observable. Thinking about it, as long as you're not using instant loading, you should also be able to just drop it where you like.

It's probably a good idea to dive into the code base and understand how Material for MkDocs works internally. It's not that complicated, but some understanding of observables and RxJS is certainly helpful. I'm sorry, that I cannot assist you getting this to work. My (spare) time is limited, so I can only give you some hints where to look.

> On another note, I was trying to implement your original suggestion of just having one search bar and positioning it based on media queries, but was running into z fighting with the header if it exists outside that element. I think we're hoping to take advantage of some of the special behaviors (taking up the full screen for example) when in mobile without re-inventing all of the associated CSS.

Welcome to the pain of CSS 😅

**Reply by Mendanbar** [reply]
I totally understand.  You've given us some really good hints, so hopefully we'll be able to make it work. 🤞 

Thanks!
